src/providers/cryptoauthlib/mod.rs

```diff
@@ -199,6 +208,16 @@ impl Provider {
             _ => None,
         }
     }
+
+    // Get the deployment specific write key. With this the keys can be stored encrypted in their slots.
+    // For ECC private keys it is obligatory, for Aes it is an option.
+    fn get_write_encrypt_key(&self) -> [u8; rust_cryptoauthlib::ATCA_KEY_SIZE] {
+        [
+            0x4D, 0x50, 0x72, 0x6F, 0x20, 0x49, 0x4F, 0x20, 0x4B, 0x65, 0x79, 0x20, 0x9E, 0x31,
+            0xBD, 0x05, 0x82, 0x58, 0x76, 0xCE, 0x37, 0x90, 0xEA, 0x77, 0x42, 0x32, 0xBB, 0x51,
+            0x81, 0x49, 0x66, 0x45,
+        ]
+    }
```

Comment:
Sorry for my lack of knowledge on cryptoauthlib, but where does this value come from?

Reply:
In my understanding, this is a completely arbitrary value. This particular one comes from rust-cryptoauthlib :-).
I think setting this write key should be adapted to a real world deployment scenario, e.g. this might be the MAC address of a network card or it can simply be stored in a configuration file, but at the moment there is just this getter.
Please note - this key is used for encrypting private keys while they are stored (imported or generated - not written) into their slots. Afterwards the private keys cannot be exported or anything like that, even encrypted. The only way to get the private key would be to (1) sniff for import messages or, (2) in case key is generated, sniff the i2c bus or to desolder the chip and try to read slot contents, which in general is pretty difficult.
Just yet another level of security.

Reply:
Ah I see. I understand that this is another level of security. I'm concerned about the confidentiality of this value though. I would think about two options:
1. We make it clear that encrypting stored keys is only done because required by the crypto auth lib but has no security value and use a dummy value for the write encrypt key (like only `0xFF`, to make it obvious it's not used.).
2. We use this value properly and make the value of that key come from the configuration file with a new field. We are already storing confidential values in the config file so there is precedent.

Is this write key also a way to prevent multiple people accessing the same ECC key on the same ATEC device, as you would need to know it if you want to use the key?

Reply:
The need for keeping this key confidential is debatable in my opinion. I suspect it exists to keep together: the cryptochip (with keys stored inside) and the piece of hardware (computer) it is soldered into.
About the key being a protection against concurrent accesses - well, it can offer kind of (poor) protection, but if the slot is configured to be writable, its contents can be easily overwritten. And if one user configures the write key, the write key is non-permanently stored on chip and another user with the access to this particular i2c bus could use the slots as if they were not encrypted. 
I discussed how can protection be ensured and it needs carefully designed hardware, where there is a dedicated i2c bus, exclusively assigned to parsec user by means of udev settings. Then the parsec user will own everything connected via this bus, but at least nobody else will be able to "disturb".

Reply:
Alright, thanks for the explanation! The assumption is that no-one will be able to read/write on the i2c bus where the chip is.